### PR TITLE
[HOPSWORKS-1538] add a function to automatically set up the hopsfs an…

### PR DESCRIPTION
**Pull request by berthoug**
…d hive clients on a databricks cluster

https://logicalclocks.atlassian.net/browse/HOPSWORKS-1538

### CODE REVIEW

File: hops/featurestore_impl/rest/rest_rpc.py [hunk 1]
```diff
@@ -6,6 +6,30 @@
 from hops.exceptions import RestAPIError
 import json
 
+# To be compatible with python2 and python3
```
Comment (steffengr) [inline hops/featurestore_impl/rest/rest_rpc.py]: Are we still Python 2 compatible?

Comment (berthoug) [reply hops/featurestore_impl/rest/rest_rpc.py]: No, I copied this part from hopsworks-cloud-sdk, I can probably remove it.

File: hops/featurestore.py [hunk 2]
```diff
@@ -1885,17 +1887,117 @@ def connect(host, project_name, port = 443, region_name = constants.AWS.DEFAULT_
...
+    spark.hadoop.client.rpc.ssl.enabled.protocol TLSv1.2
+    spark.hadoop.hops.ssl.keystores.passwd.name DBFS_FOLDER/material_passwd
+    spark.hadoop.hops.ssl.keystore.name DBFS_FOLDER/keyStore.jks
+    spark.hadoop.hops.ssl.trustore.name DBFS_FOLDER/trustStore.jks
+    spark.sql.hive.metastore.jars /tmp/apache-hive-bin/lib/*
+    spark.hadoop.hive.metastore.uris thrift://HOST:9083
+
+    Then save and restart the cluster.
+    """
+    instructions = instructions.replace("PATH_ARG", cert_folder)
```
[the remark this answers is not in the document]
Comment (berthoug) [reply hops/featurestore.py]: done

File: hops/featurestore_impl/rest/rest_rpc.py (hunk 1 above)
Comment (berthoug) [reply hops/featurestore_impl/rest/rest_rpc.py]: done

File: hops/featurestore.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (berthoug) [reply hops/featurestore.py]: done

File: hops/featurestore.py [hunk 3]
```diff
@@ -1885,17 +1887,117 @@ def connect(host, project_name, port = 443, region_name = constants.AWS.DEFAULT_
...
+
+    #get the credential and save them in the dbfs folder
+    credentials = rest_rpc._get_credentials(project_id)
+    util.write_b64_cert_to_bytes(str(credentials['kStore']), path=os.path.join(dbfs_folder, 'keyStore.jks'))
+    util.write_b64_cert_to_bytes(str(credentials['tStore']), path=os.path.join(dbfs_folder, 'trustStore.jks'))
+    #The hive connector need to read the password at cluster startup so we need to write it in a file
+    with open(os.path.join(dbfs_folder, 'material_passwd'), 'w') as f:
+        f.write(str(credentials['password']))
+
+    #get the libraries and save them in the dbfs folder
```
[the remark this answers is not in the document]
Comment (berthoug) [reply hops/featurestore.py]: done

File: hops/featurestore.py [hunk 4]
```diff
@@ -1885,17 +1887,117 @@ def connect(host, project_name, port = 443, region_name = constants.AWS.DEFAULT_
...
+
+    project_info = rest_rpc._get_project_info(project_name)
+    project_id = str(project_info['projectId'])
+    os.environ[constants.ENV_VARIABLES.HOPSWORKS_PROJECT_ID_ENV_VAR] = project_id
+
+    Path(dbfs_folder + "/scripts").mkdir(parents=True, exist_ok=True)
+
+    #get the credential and save them in the dbfs folder
+    credentials = rest_rpc._get_credentials(project_id)
+    util.write_b64_cert_to_bytes(str(credentials['kStore']), path=os.path.join(dbfs_folder, 'keyStore.jks'))
```
[the remark this answers is not in the document]
Comment (berthoug) [reply hops/featurestore.py]: done

File: hops/util.py [hunk 5]
```diff
@@ -72,7 +75,7 @@ def _get_hopsworks_rest_endpoint():
         The hopsworks REST endpoint for making requests to the REST API
 
     """
-    return os.environ[constants.ENV_VARIABLES.REST_ENDPOINT_END_VAR]
+    return 'https://' + os.environ[constants.ENV_VARIABLES.REST_ENDPOINT_END_VAR]
```
Comment (SirOibaf) [inline hops/util.py]: In Hopsworks the environment variable already contain `https://`

File: hops/featurestore.py [hunk 6]
```diff
@@ -1885,17 +1887,144 @@ def connect(host, project_name, port = 443, region_name = constants.AWS.DEFAULT_
...
+    Set up the hopfs and hive connector on a detabricks cluster
+
+    Example usage:
+
+    >>> featurestore.setup_databricks("hops.site", "my_feature_store", "/dbfs/mnt/certs")
+
+    Args:
+        :host: the hostname of the Hopsworks cluster
+        :project_name: the name of the project hosting the feature store to be used
+        :dbfs_folder: the folder in dbfs in which to store the Hopsworks certificates and the libraries to be installed when \
```
Comment (SirOibaf) [inline hops/featurestore.py]: the documentation doesn't match the parameter name.

File: hops/featurestore.py [hunk 7]
```diff
@@ -1885,17 +1887,144 @@ def connect(host, project_name, port = 443, region_name = constants.AWS.DEFAULT_
...
+        the cluster restart
+        :port: the REST port of the Hopsworks cluster
+        :region_name: The name of the AWS region in which the required secrets are stored
+        :secrets_store: The secrets storage to be used. Either secretsmanager or parameterstore.
+
+    Returns:
+        None
+    """
+    
+    dbfs_folder = "/dbfs/" + cert_folder
```
Comment (SirOibaf) [inline hops/featurestore.py]: Is `/dbfs/`  transparent to the user? Or will the user expect to pass `/dbfs/` as well?

Comment (SirOibaf) [reply hops/featurestore.py]: actually in the example you already have `dbfs` in the parameter

File: hops/featurestore.py (hunk 6 above)
Comment (berthoug) [reply hops/featurestore.py]: will fix it

File: hops/featurestore.py (hunk 7 above)
Comment (berthoug) [reply hops/featurestore.py]: The problem is that it is a mounted folder, so the path in dbfs is the path given by the cert_folder argument, but the path in the machine has to start with /dbfs/

File: hops/util.py (hunk 5 above)
Comment (berthoug) [reply hops/util.py]: Ok, now I get what you meant by "in hopsworks", I fixed it.

File: hops/featurestore.py (hunk 6 above)
Comment (berthoug) [reply hops/featurestore.py]: fixed

File: hops/featurestore.py (hunk 7 above)
Comment (berthoug) [reply hops/featurestore.py]: fixed the example

**Pull request merged**